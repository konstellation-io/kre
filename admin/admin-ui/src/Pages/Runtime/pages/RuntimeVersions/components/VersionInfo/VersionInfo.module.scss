@import 'Styles/mixins';
@import 'Styles/variables';
@import 'Styles/funcs';
@import 'Styles/colors';

$color-border: palette(base, 400);

.container {
  display: flex;
  flex-direction: row;
  border-bottom: solid 1px $color-border;
  transition: background-color ease 0.3s;
  color: font-color(light);

  &.even {
    background-color: $bg-color-lighter;
  }

  .creationDate,
  .name,
  .version,
  .statusLine,
  .statusCircle,
  .description {
    transition: color ease 0.3s, border ease 0.3s, background-color ease 0.3s;
  }

  &:hover {
    background-color: $color-black;
    cursor: pointer;

    .creationDate,
    .name,
    .version,
    .description {
      color: font-color(light);
    }

    .statusLine {
      border-bottom-color: $color-ok;
      border-top-color: $color-ok;
    }

    .statusCircle {
      background-color: $color-ok;
    }
  }

  p {
    margin: 0;
  }
}

.col1 {
  padding: 20px 16px 40px 40px;
  flex-grow: 1;
  max-width: 400px;
  min-width: 340px;
  width: 50%;
  display: flex;
  flex-direction: row;

  .labelContainer {
    .label {
      @include font-button;
      color: $bg-color;
      text-align: center;
      line-height: 24px;

      width: 96px;
      height: 24px;
      margin-right: 24px;

      @each $state in $version-statuses {
        &.#{$state} {
          background-color: get-color-prop($state);
        }
      }
    }
  }

  .actionsContainer {
    width: -webkit-fill-available;
  }

  .creation {
    @include font-caption;
    font-weight: 700;
    line-height: 24px;

    display: flex;
    flex-direction: row;
    align-items: center;
    white-space: nowrap;
    margin-left: auto;
    width: fit-content;
    margin-bottom: 20px;
  }

  .creationDate {
    margin-left: 5px;
  }

  .actionButtons {
    width: fit-content;
    margin-left: auto;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;

    > div {
      margin-left: auto;
    }
  }
}

.col2 {
  width: 50%;
  padding: 20px 20px 40px 30px;
  border-left: solid 1px $color-border;

  .name {
    @include font-body;
    font-weight: 600;
    line-height: 17px;

    margin-bottom: 6px;
  }

  .version {
    @include font-caption;
    line-height: 24px;

    margin-bottom: 15px;
  }

  .descriptionTitle,
  .activatedTitle {
    @include font-caption;
    color: font-color(regular);

    margin-bottom: 8px;
  }

  .activatedTitle {
    margin-top: 15px;
  }

  .col2CreatorName,
  .col2CreationDate {
    display: flex;
    flex-direction: row;
    align-items: center;
  }

  .description {
    @include font-small;
  }

  .activatedContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
  }

  .activatedAuthor,
  .activatedDate {
    @include font-caption;
    line-height: 17px;

    margin: 0 8px 0 8px;
  }
}
