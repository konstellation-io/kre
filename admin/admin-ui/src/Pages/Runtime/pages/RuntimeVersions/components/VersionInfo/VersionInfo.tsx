import CalendarIcon from '@material-ui/icons/Today';
import EmailIcon from '@material-ui/icons/Email';
import { GetVersionConfStatus_versions } from 'Graphql/queries/types/GetVersionConfStatus';
import { Link } from 'react-router-dom';
import ROUTE from 'Constants/routes';
import React from 'react';
import TimeIcon from '@material-ui/icons/AccessTime';
import { VersionStatus } from 'Graphql/types/globalTypes';
import { buildRoute } from 'Utils/routes';
import cx from 'classnames';
import { formatDate } from 'Utils/format';
import styles from './VersionInfo.module.scss';
import { useParams } from 'react-router';

type Props = {
  version: GetVersionConfStatus_versions;
};
function VersionInfo({ version }: Props) {
  const { runtimeId } = useParams();

  const isPublishedVersion = version.status === VersionStatus.PUBLISHED;
  const hasErrors = version.status === VersionStatus.ERROR;

  const versionPath = buildRoute.version(
    ROUTE.RUNTIME_VERSION_STATUS,
    runtimeId,
    version.id
  );

  const errors = version.errors.map((error: string, index: number) => (
    <p key={index} className={styles.versionError}>
      {error}
    </p>
  ));
  return (
    <Link to={versionPath}>
      <div className={styles.container} id={`versionInfoElement_${version.id}`}>
        <div className={styles.col1}>
          <div className={styles.labelContainer}>
            <div className={cx(styles.label, styles[version.status])}>
              {version.status}
            </div>
          </div>
          <div className={styles.actionsContainer}>
            <div className={styles.creation}>
              <CalendarIcon className="icon-small" />
              <p className={styles.creationDate}>
                {`CREATED: ${formatDate(new Date(version.creationDate))}`}
              </p>
            </div>
          </div>
        </div>
        <div className={styles.col2}>
          <p className={styles.version}>{`VERSION ${version.name}`}</p>
          {version.description && (
            <p className={styles.descriptionTitle}>DESCRIPTION</p>
          )}
          <p className={styles.description}>{version.description}</p>
          {isPublishedVersion && (
            <>
              <p className={styles.activatedTitle}>PUBLISHED BY</p>
              <div className={styles.activatedContainer}>
                <div className={styles.col2CreatorName}>
                  <EmailIcon className="icon-small" />
                  <span className={styles.activatedAuthor}>
                    {version &&
                      version.publicationAuthor &&
                      version.publicationAuthor.email}
                  </span>
                </div>
                <div className={styles.col2CreationDate}>
                  <TimeIcon className="icon-small" />
                  <span className={styles.activatedDate}>
                    {version.publicationDate}
                  </span>
                </div>
              </div>
            </>
          )}
          {hasErrors && (
            <>
              <p className={styles.descriptionTitle}>ERRORS</p>
              <div>{errors}</div>
            </>
          )}
        </div>
      </div>
    </Link>
  );
}

export default VersionInfo;
