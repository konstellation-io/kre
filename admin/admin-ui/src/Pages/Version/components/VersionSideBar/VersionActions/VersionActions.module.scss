@import 'Styles/colors';
@import 'Styles/mixins';

.wrapper {
  color: font-color(light);
  display: flex;
  flex-direction: row;
  background-color: #26292c;
  align-items: center;
  margin-top: auto;

  &.quickActions {
    background-color: transparent;
  }

  .item {
    @include font-body;
    font-weight: 600;

    height: 48px;
    padding-left: 16px;
    display: flex;
    flex-direction: row;
    align-items: center;
    flex-grow: 1;
    cursor: pointer;

    .icon {
      margin-right: 8px;
      line-height: 0;
    }

    &:hover {
      color: font-color(light);
    }
  }
}

.disabled {
  cursor: default;
  pointer-events: none;
  color: font-color(regular);
}

.warning {
  @include font-small;
  color: palette(alert);

  margin-bottom: $grid-unit;

  .publishedVersion {
    text-transform: uppercase;
    color: font-color(light);
  }
}
