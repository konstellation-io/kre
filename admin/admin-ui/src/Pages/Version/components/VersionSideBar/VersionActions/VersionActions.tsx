import {
  BUTTON_THEMES,
  Button,
  CHECK,
  ModalContainer,
  ModalLayoutInfo,
  ModalLayoutJustify
} from 'kwc';
import {
  GetVersionConfStatus_runtime,
  GetVersionConfStatus_versions
} from 'Graphql/queries/types/GetVersionConfStatus';
import React, { FunctionComponent, useEffect, useRef, useState } from 'react';
import Tag, { TagTypes } from 'Components/Tag/Tag';

import { Link } from 'react-router-dom';
import PublishIcon from '@material-ui/icons/Publish';
import ROUTE from 'Constants/routes';
import StartIcon from '@material-ui/icons/PlayCircleOutline';
import StopIcon from '@material-ui/icons/PauseCircleFilled';
import { SvgIconProps } from '@material-ui/core/SvgIcon';
import UnpublishIcon from '@material-ui/icons/GetApp';
import { VersionStatus } from 'Graphql/types/globalTypes';
import { buildRoute } from 'Utils/routes';
import cx from 'classnames';
import { get } from 'lodash';
import styles from './VersionActions.module.scss';
import { useForm } from 'react-hook-form';
import useVersionAction from '../../../utils/hooks';

function verifyComment(value: string) {
  return CHECK.getValidationError([CHECK.isFieldNotEmpty(value)]);
}

type WarningProps = {
  runtime: GetVersionConfStatus_runtime;
  publishedVersion: GetVersionConfStatus_versions;
};
function Warning({ runtime, publishedVersion }: WarningProps) {
  return (
    <>
      <Tag type={TagTypes.WARNING}>WARNING</Tag>A published version already
      exists, publishing this version will unpublish the following version:{' '}
      <Link
        to={buildRoute.version(
          ROUTE.RUNTIME_VERSION,
          runtime.id,
          publishedVersion.id
        )}
      >
        <span
          className={styles.publishedVersion}
          title={publishedVersion.description}
        >
          {publishedVersion.name}
        </span>
      </Link>
    </>
  );
}

type ModalInfo = {
  action: (comment: string) => void;
  label: string;
};

type ActionProps = {
  Icon: FunctionComponent<SvgIconProps>;
  label: string;
  action?: Function;
  disabled?: boolean;
  primary?: boolean;
};

type FormData = {
  comment: string;
};

type Props = {
  runtime: GetVersionConfStatus_runtime;
  versions: GetVersionConfStatus_versions[];
  version: GetVersionConfStatus_versions;
  quickActions?: boolean;
};

function VersionActions({
  runtime,
  versions,
  version,
  quickActions = false
}: Props) {
  const { handleSubmit, setValue, register, unregister, errors } = useForm<
    FormData
  >({
    defaultValues: {
      comment: ''
    }
  });
  const [showConfirmation, setShowConfirmation] = useState(false);
  const modalInfo = useRef<ModalInfo>({
    action: () => {},
    label: ''
  });

  useEffect(() => {
    register('comment', { validate: verifyComment });

    return () => unregister('comment');
  }, [register, unregister, setValue]);

  function onSubmit() {
    handleSubmit((formData: FormData) =>
      modalInfo.current.action(formData.comment)
    )();
  }

  const {
    publishVersion,
    startVersion,
    stopVersion,
    unpublishVersion,
    getMutationVars
  } = useVersionAction(runtime.id);

  function openModal(label: string, mutation: Function) {
    modalInfo.current = {
      action: (comment: string) => {
        mutation(getMutationVars(version.id, comment));
        closeModal();
      },
      label
    };
    setShowConfirmation(true);
  }

  function closeModal() {
    setShowConfirmation(false);
  }

  let buttons: ActionProps[] = [];

  switch (version.status) {
    case VersionStatus.CREATED:
    case VersionStatus.STOPPED:
      buttons[0] = {
        Icon: StartIcon,
        label: 'START',
        action: () => openModal('START', startVersion),
        primary: version.config.completed,
        disabled: !version.config.completed
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        disabled: true
      };
      break;
    case VersionStatus.STOPPING:
      buttons[0] = {
        Icon: StartIcon,
        label: 'START',
        action: () => openModal('START', startVersion),
        primary: version.config.completed,
        disabled: true
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        disabled: true
      };
      break;
    case VersionStatus.CREATING:
      buttons[0] = {
        Icon: StartIcon,
        label: 'START',
        disabled: true
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        disabled: true
      };
      break;
    case VersionStatus.STARTING:
      buttons[0] = {
        Icon: StopIcon,
        label: 'STOP',
        action: () => openModal('STOP', stopVersion),
        disabled: true
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        disabled: true
      };
      break;
    case VersionStatus.STARTED:
      buttons[0] = {
        Icon: StopIcon,
        label: 'STOP',
        action: () => openModal('STOP', stopVersion)
      };
      buttons[1] = {
        Icon: PublishIcon,
        label: 'PUBLISH',
        action: () => openModal('PUBLISH', publishVersion),
        primary: true
      };
      break;
    case VersionStatus.PUBLISHED:
      buttons[0] = {
        Icon: StopIcon,
        label: 'STOP',
        disabled: true
      };
      buttons[1] = {
        Icon: UnpublishIcon,
        label: 'UNPUBLISH',
        action: () => openModal('UNPUBLISH', unpublishVersion),
        primary: true
      };
      break;
  }

  if (quickActions) {
    buttons = buttons.filter(button => !button.disabled);
  }

  const publishedVersion = versions.filter(
    v => v.status === VersionStatus.PUBLISHED
  )[0];

  const showWarning = publishedVersion && modalInfo.current.label === 'PUBLISH';

  return (
    <div
      className={cx(styles.wrapper, { [styles.quickActions]: quickActions })}
    >
      {buttons.map(btn => (
        <Button
          key={btn.label}
          onClick={() => btn.action && btn.action()}
          label={quickActions ? '' : btn.label}
          title={quickActions ? btn.label : ''}
          Icon={btn.Icon}
          iconSize="icon-regular"
          theme={
            quickActions ? BUTTON_THEMES.TRANSPARENT : BUTTON_THEMES.DEFAULT
          }
          disabled={btn.disabled}
          primary={quickActions ? false : btn.primary}
          style={{ flexGrow: 1 }}
        />
      ))}
      {showConfirmation && (
        <ModalContainer
          title={`YOU ARE ABOUT TO ${modalInfo.current.label} A VERSION`}
          actionButtonLabel={modalInfo.current.label}
          onAccept={onSubmit}
          onCancel={closeModal}
          warning={showWarning}
          blocking
        >
          <ModalLayoutJustify
            onUpdate={(value: string) => setValue('comment', value)}
            submit={onSubmit}
            error={get(errors.comment, 'message', '') as string}
          />
          {showWarning && (
            <ModalLayoutInfo className={styles.warning}>
              <Warning runtime={runtime} publishedVersion={publishedVersion} />
            </ModalLayoutInfo>
          )}
        </ModalContainer>
      )}
    </div>
  );
}

export default VersionActions;
