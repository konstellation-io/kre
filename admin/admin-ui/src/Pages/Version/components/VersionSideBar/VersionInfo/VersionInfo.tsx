import { GetVersionConfStatus_versions } from 'Graphql/queries/types/GetVersionConfStatus';
import React from 'react';
import StatusCircle from 'Components/StatusCircle/StatusCircle';
import cx from 'classnames';
import styles from './VersionInfo.module.scss';

type VersionListItemProps = {
  version: GetVersionConfStatus_versions;
};

function VersionInfo({ version }: VersionListItemProps) {
  return (
    <div className={styles.wrapper}>
      <div className={styles.name}>
        <div>{version.name}</div>
      </div>
      <div className={styles.labelContainer}>
        <StatusCircle status={version.status} />
        <div className={cx(styles.label, styles[version.status])}>
          {version.status}
        </div>
      </div>

      <div>
        <div className={styles.desc}>{version.description}</div>
      </div>
    </div>
  );
}

export default VersionInfo;
