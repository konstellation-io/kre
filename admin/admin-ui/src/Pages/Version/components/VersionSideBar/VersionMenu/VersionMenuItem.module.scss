@import 'Styles/colors';
@import 'Styles/mixins';

$color-active: palette(lowlight, 900);
$color-hover: palette(lowlight, 700);

.item {
  @include font-button;
  font-weight: 700;
  color: font-color(lowlight);

  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 16px;
  height: $grid-unit * 7;

  transition: background-color ease 0.4s, color ease 0.4s;

  .icon {
    margin-right: 16px;
  }

  .iconWarning {
    margin-left: auto;
    color: $color-warn;
  }

  &:hover {
    color: font-color(light);
    background-color: $color-hover;
  }
}

.active {
  cursor: default;
  pointer-events: none;
  color: font-color(light);

  .item {
    background-color: $color-active;
    color: font-color(light);
  }
}
