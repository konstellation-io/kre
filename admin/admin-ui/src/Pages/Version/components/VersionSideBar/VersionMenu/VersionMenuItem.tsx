import React, { FunctionComponent } from 'react';

import IconWarning from '@material-ui/icons/Warning';
import { NavLink } from 'react-router-dom';
import { SvgIconProps } from '@material-ui/core/SvgIcon';
import cx from 'classnames';
import styles from './VersionMenuItem.module.scss';

export type VersionMenuItemProps = {
  label: string;
  to: string;
  exact?: boolean;
  warning?: string;
  Icon: FunctionComponent<SvgIconProps>;
};

function VersionMenuItem({
  label,
  to,
  exact,
  warning,
  Icon
}: VersionMenuItemProps) {
  const isExternal = /^https?:\/\/.+$/.test(to);
  const itemContent = (
    <div className={styles.item}>
      <div className={styles.icon}>
        <Icon className="icon-regular" />
      </div>
      <span>{label}</span>
      {warning && (
        <div title={warning} className={styles.iconWarning}>
          <IconWarning className={cx('icon-regular', styles.warning)} />
        </div>
      )}
    </div>
  );
  return isExternal ? (
    <a href={to} target="_blank">
      {itemContent}
    </a>
  ) : (
    <NavLink to={to} activeClassName={styles.active} exact={exact} replace>
      {itemContent}
    </NavLink>
  );
}

export default VersionMenuItem;
