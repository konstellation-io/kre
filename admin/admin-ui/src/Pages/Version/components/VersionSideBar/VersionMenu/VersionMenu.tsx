import {
  GetVersionConfStatus_runtime,
  GetVersionConfStatus_versions
} from 'Graphql/queries/types/GetVersionConfStatus';
import VersionMenuItem, { VersionMenuItemProps } from './VersionMenuItem';

import IconDeviceHub from '@material-ui/icons/DeviceHub';
import IconDoc from '@material-ui/icons/Toc';
import IconSettings from '@material-ui/icons/Settings';
import IconShowChart from '@material-ui/icons/ShowChart';
import ROUTE from 'Constants/routes';
import React from 'react';
import { buildRoute } from 'Utils/routes';
import { checkPermission } from 'rbac-rules';
import styles from './VersionMenu.module.scss';
import useUserAccess from 'Hooks/useUserAccess';

type VersionDetailsProps = {
  runtime: GetVersionConfStatus_runtime;
  version: GetVersionConfStatus_versions;
};

function VersionMenu({ runtime, version }: VersionDetailsProps) {
  const { accessLevel } = useUserAccess();

  const itemProps: VersionMenuItemProps[] = [
    {
      label: 'STATUS',
      to: ROUTE.RUNTIME_VERSION_STATUS,
      exact: false,
      Icon: IconDeviceHub
    }
  ];

  if (version.hasDoc) {
    itemProps.push({
      label: 'DOCUMENTATION',
      to: ROUTE.RUNTIME_VERSION_DOCUMENTATION,
      exact: false,
      Icon: IconDoc
    });
  }

  itemProps.push({
    label: 'PREDICTIONS',
    to: ROUTE.RUNTIME_VERSION_METRICS,
    Icon: IconShowChart
  });

  itemProps.push({
    label: 'MEASUREMENTS',
    to: runtime.measurementsUrl,
    Icon: IconShowChart
  });

  if (checkPermission(accessLevel, 'version:edit')) {
    itemProps.push({
      label: 'CONFIGURATION',
      to: ROUTE.RUNTIME_VERSION_CONFIGURATION,
      Icon: IconSettings,
      warning: version.config.completed ? '' : 'Configuration is not completed'
    });
  }

  itemProps.forEach(p => {
    p.to = buildRoute.version(p.to, runtime.id, version.id);
  });

  const items = itemProps.map(props => (
    <VersionMenuItem key={`${props.label}-${props.to}`} {...props} />
  ));

  return <div className={styles.wrapper}>{items}</div>;
}

export default VersionMenu;
