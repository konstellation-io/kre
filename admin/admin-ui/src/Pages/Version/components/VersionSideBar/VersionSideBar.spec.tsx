import { runtime, version } from 'Mocks/version';

import React from 'react';
import VersionActions from './VersionActions/VersionActions';
import VersionInfo from './VersionInfo/VersionInfo';
import VersionMenu from './VersionMenu/VersionMenu';
import VersionSideBar from './VersionSideBar';
import { shallow } from 'enzyme';

jest.mock('@apollo/client', () => ({
  useMutation: jest.fn(() => [jest.fn(), { loading: false }])
}));

describe('VersionSideBar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <VersionSideBar version={version} runtime={runtime} versions={[]} />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show right components', async () => {
    expect(wrapper.exists(VersionInfo)).toBeTruthy();
    expect(wrapper.exists(VersionMenu)).toBeTruthy();
    expect(wrapper.exists(VersionActions)).toBeTruthy();
    expect('a').toBeTruthy();
  });

  it('do now show when there is no runtime', async () => {
    wrapper.setProps({ runtime: undefined });
    expect(wrapper.find(VersionSideBar).exists('.wrapper')).toBeFalsy();
  });

  it('do now show when there is no version', async () => {
    wrapper.setProps({ version: undefined });
    expect(wrapper.find(VersionSideBar).exists('.wrapper')).toBeFalsy();
  });
});
