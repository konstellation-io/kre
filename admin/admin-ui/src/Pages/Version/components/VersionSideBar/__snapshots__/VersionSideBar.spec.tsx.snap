// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionSideBar matches snapshot 1`] = `
<div
  className="wrapper"
>
  <Link
    to="/runtimes/runtimeId"
  >
    <div
      className="backSection"
    >
      <KeyboardBackspaceIcon
        className="icon-regular"
      />
      <div
        className="runtimeHexagon"
      >
        <RuntimeHexagon
          runtime={
            Object {
              "__typename": "Runtime",
              "creationAuthor": Object {
                "email": "some@user.com",
              },
              "description": "Some Description",
              "id": "runtimeId",
              "measurementsUrl": "measurementsUrl",
              "name": "runtimeName",
              "status": "STARTED",
            }
          }
          size={18}
        />
      </div>
      <div
        className="runtimeName"
      >
        runtimeName
      </div>
    </div>
  </Link>
  <div
    className="runtimeNameSection"
  />
  <VersionInfo
    version={
      Object {
        "__typename": "Version",
        "config": Object {
          "completed": false,
          "vars": Array [],
        },
        "creationAuthor": Object {
          "__typename": "User",
          "email": "user@domain.com",
          "id": "userId",
        },
        "creationDate": "2020/01/01",
        "description": "versionDescription",
        "errors": Array [],
        "hasDoc": false,
        "id": "versionId",
        "name": "versionName",
        "publicationAuthor": Object {
          "__typename": "User",
          "email": "user@domain.com",
          "id": "userId",
        },
        "publicationDate": "2020/01/01",
        "status": "STARTED",
      }
    }
  />
  <VersionMenu
    runtime={
      Object {
        "__typename": "Runtime",
        "creationAuthor": Object {
          "email": "some@user.com",
        },
        "description": "Some Description",
        "id": "runtimeId",
        "measurementsUrl": "measurementsUrl",
        "name": "runtimeName",
        "status": "STARTED",
      }
    }
    version={
      Object {
        "__typename": "Version",
        "config": Object {
          "completed": false,
          "vars": Array [],
        },
        "creationAuthor": Object {
          "__typename": "User",
          "email": "user@domain.com",
          "id": "userId",
        },
        "creationDate": "2020/01/01",
        "description": "versionDescription",
        "errors": Array [],
        "hasDoc": false,
        "id": "versionId",
        "name": "versionName",
        "publicationAuthor": Object {
          "__typename": "User",
          "email": "user@domain.com",
          "id": "userId",
        },
        "publicationDate": "2020/01/01",
        "status": "STARTED",
      }
    }
  />
  <Can
    perform="version:edit"
  >
    <VersionActions
      runtime={
        Object {
          "__typename": "Runtime",
          "creationAuthor": Object {
            "email": "some@user.com",
          },
          "description": "Some Description",
          "id": "runtimeId",
          "measurementsUrl": "measurementsUrl",
          "name": "runtimeName",
          "status": "STARTED",
        }
      }
      version={
        Object {
          "__typename": "Version",
          "config": Object {
            "completed": false,
            "vars": Array [],
          },
          "creationAuthor": Object {
            "__typename": "User",
            "email": "user@domain.com",
            "id": "userId",
          },
          "creationDate": "2020/01/01",
          "description": "versionDescription",
          "errors": Array [],
          "hasDoc": false,
          "id": "versionId",
          "name": "versionName",
          "publicationAuthor": Object {
            "__typename": "User",
            "email": "user@domain.com",
            "id": "userId",
          },
          "publicationDate": "2020/01/01",
          "status": "STARTED",
        }
      }
      versions={Array []}
    />
  </Can>
</div>
`;
